### PR TITLE
Raspberry sensors

### PR DESCRIPTION
**Pull request by crcro**
detect linux that runs on pi boxes and adds sensors (requires librenms-agent PR #48)
screen shot 2016-08-14 at 5 53 31 pm

### CODE REVIEW

**Comment by Rosiak** [comment]
Don't think the "disabled" should be critital in the state part.

**Comment by crcro** [comment]
@Rosiak better to warning?

**Comment by laf** [comment]
You can label it as disabled but you should set it to the unknown value (3)

Great work though :)

File: doc/Extensions/Applications.md [hunk 1]
```diff
@@ -168,6 +168,23 @@ extend memcached /usr/local/bin/memcached
...
+##### Extend SNMP
+1. Copy the [raspberry script](https://github.com/librenms/librenms-agent/blob/master/snmp/raspberry.sh) to `/opt/` (or any other suitable location) on your PI host.
+2. Make the script executable: `chmod +x /opt/raspberry.sh`
+3. Edit your snmpd.conf file (usually `/etc/snmp/snmpd.conf`) and add:
+```
+extend raspberry /opt/raspberry.sh
+```
+4. Edit your sudo users (usually `visudo`) and add at the bottom:
+```
+snmp ALL=(ALL) NOPASSWD: ALL
```
Comment (laf) [inline doc/Extensions/Applications.md]: Is this a good idea? Not better to allow the specific command - or does net-snmp on the Pi not run as root anyway?

Comment (crcro) [inline doc/Extensions/Applications.md]: on the raspberry pi that i had access to (running debian 7.8) it runs as snmp user

Comment (laf) [inline doc/Extensions/Applications.md]: Ok so just restrict the command that snmp can run with no password.

No function to convert between those formats I don't believe.

**Comment by laf** [comment]
Sorry @crcro - needs a rebase :(

**Comment by crcro** [comment]
rebased

**Comment by arrmo** [comment]
Hi,

Likely me just being dumb (OK, I'm sure it is unfortunately), but struggling with this a bit. I set up the script file, and it runs nicely on the Pi. I configured snmpd.conf and visudo, restarted, and I do get output from walking snmp, but it's as follows,

NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".1 = STRING:
NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".2 = STRING: We trust you have received the usual lecture from the local System
NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".3 = STRING: Administrator. It usually boils down to these three things:
NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".4 = STRING:
NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".5 = STRING:     #1) Respect the privacy of others.
NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".6 = STRING:     #2) Think before you type.
NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".7 = STRING:     #3) With great power comes great responsibility.
NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".8 = STRING:
NET-SNMP-EXTEND-MIB::nsExtendOutLine."raspberry".9 = STRING: sudo: no tty present and no askpass program specified

This repeats several times (OK, you get the picture ... :)). Thoughts?

Thanks!

**Comment by crcro** [comment]
hi, could you please try commenting the visudo line:

Defaults requiretty

**Comment by arrmo** [comment]
Do you mean adding this to the end of the line? Tried that, no joy ... :(

Pulling my hair out (and I'm short of that ... LOL). From the command line, if I execute "sudo raspberry.sh" -> no issue at all, works fine.

Thanks!

**Comment by crcro** [comment]
try changing:

snmp ALL=(ALL) NOPASSWD: /opt/raspberry.sh to
snmp ALL=(ALL) NOPASSWD: ALL

and give it a try

**Comment by arrmo** [comment]
Joy - thanks!

Not sure why this is needed, but no biggie.

Much appreciated!

**Comment by crcro** [comment]
could you please give it a try with the following:

snmp ALL=(ALL) NOPASSWD: /opt/raspberry.sh, /usr/bin/vcgencmd*

**Comment by arrmo** [comment]
Yep, that works! And I tried removing /usr/bin/vcgencmd* => breaks. So this is key.

Thanks!

**Comment by cweakland** [comment]
First off, thanks for all the great work. However, I have the same issue. I did a fresh install of librenms last week. I have two Pi's on the network and for both of them I see the codec charts created but I do not get the voltage/temp charts. Please let me know if I can provide any additional detail.

**Comment by mirino75** [comment]
write in this thread:
https://community.librenms.org/t/raspberry-pi-temperature-voltage-freq/430

**Pull request merged**